### PR TITLE
Add multi arch support for smartctl

### PR DESCRIPTION
**Pull request by zxhdaze**
Add support for arm64|ppc64le|riscv64|s390x
Arch name reference: https://wiki.debian.org/ArchitectureSpecificsMemo?action=recall&rev=219#fndef-28921cada1db0fe0680e9e948d3015a95dafcfbd-21

### CODE REVIEW

File: tests/unit/test_hw_tools.py [hunk 1]
```diff
@@ -764,6 +765,24 @@ def test_check(self, mock_check_deb_method):
...
+    }
+
+    # Test for unsupported architecture
+    mock_machine.return_value = "unsupported_arch"
+    try:
+        get_release_url(base_url, releases)
+    except ValueError as e:
+        assert str(e) == "Unsupported architecture: unsupported_arch"
+    else:
+        assert False, "ValueError not raised"
```
Comment (samuelallan72) [inline tests/unit/test_hw_tools.py:783]: Pytest has a [mechanism for testing this pattern](https://docs.pytest.org/en/stable/reference/reference.html#pytest-raises):

```suggestion
    with pytest.raises(ValueError, match="Unsupported architecture: unsupported_arch"):
        get_release_url(base_url, releases)
```

:)

**Review by samuelallan72 — CHANGES_REQUESTED** [review]
Thanks, looks good to me.  I left a comment about improving the test. :)

File: src/hw_tools.py [hunk 2]
```diff
@@ -411,16 +412,34 @@ def check(self) -> bool:
...
-        "https://github.com/prometheus-community/"
-        "smartctl_exporter/releases/download/v0.12.0/smartctl_exporter-0.12.0.linux-amd64.tar.gz"
+
+    _base_url = (
+        "https://github.com/prometheus-community/smartctl_exporter/releases/download/v0.12.0/"
     )
+
+    _releases = {
+        "x86_64": "smartctl_exporter-0.12.0.linux-amd64.tar.gz",
+        "arm64": "smartctl_exporter-0.12.0.linux-arm64.tar.gz",
```
Comment (gabrielcocenza) [inline src/hw_tools.py:436]: It seems that plataform lib uses the `uname` command to get the architecture and there is no `arm64` as output.
Sources: [0](https://stackoverflow.com/questions/45125516/possible-values-for-uname-m) and [1](https://wiki.debian.org/ArchitectureSpecificsMemo?action=recall&rev=219#fndef-28921cada1db0fe0680e9e948d3015a95dafcfbd-21), so I think this should be changed to `aarch64`

File: tests/unit/test_hw_tools.py [hunk 3]
```diff
@@ -764,6 +765,24 @@ def test_check(self, mock_check_deb_method):
         mock_check_deb_method.assert_called_with("smartmontools")
 
 
+@mock.patch("platform.machine")
+def test_get_release_url(mock_machine):
```
Comment (gabrielcocenza) [inline tests/unit/test_hw_tools.py:769]: Please also add a unit test with the happy path

**Review by gabrielcocenza — CHANGES_REQUESTED** [review]

**Comment by chanchiwai-ray** [comment]
we can directly jump to smartctl snap, since it's not that difficult to implement

**Comment by chanchiwai-ray** [comment]
This has been superseded by https://github.com/canonical/hardware-observer-operator/pull/327 (By the way, thanks for the efforts @zxhdaze)

**Pull request closed without merging**